### PR TITLE
fix: check if keycloak groups exist and improve logging

### PR DESCRIPTION
**Pull request by nanu-c**
# Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Description

**IMPORTANT TO KNOW**

**CHANGES**

* This adds the customer to the correct subgroup.
* Add a lot of keycloak tests
* loads env vars in keycloak via the config package
* Ensures that all keycloak groups exists on start of app
**TODO**

# Checklist:

- [x] I have commented my code (or ChatGPT did), particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings, neither in my IDE nor in my browser
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -99,6 +99,17 @@ To run a specific Test Case:
 go test ./... -p 1 -v -cover -run NameOfTestCase
 ```
 
+Test coverage:
+
+```bash
+go test ./... -p 1 -coverprofile=cover.out
```
Comment (lebe1) [inline README.md:105]: When I run this in my app directory, I receive the following error `open my_path/backend/app/.env: no such file or directory`
Do you have an env variable created there which is not being pushed by git?
Screenshot from 2023-12-21 11-31-09

**Review by lebe1 — COMMENTED** [review]
I have one final question/thought:
1. Shouldn't we add the new keycloak environment variables to our `.env.examples` like 'KEYCLOAK_CUSTOMER_GROUP' ?
Besides this, I could even see the two testing versions in our keycloak GUI being created, which is really nice! :)
image

File: README.md (hunk 1 above)
Comment (nanu-c) [reply README.md:105]: It has to be run in the same environment as the tests. So this indicates, that you have run it outside the docker container.

**Comment by nanu-c** [comment]
> I have one final question/thought:
> 
>     1. Shouldn't we add the new keycloak environment variables to our `.env.examples` like 'KEYCLOAK_CUSTOMER_GROUP' ?
>        Besides this, I could even see the two testing versions in our keycloak GUI being created, which is really nice! :)
>        image

maybe we should do this, but who will change it? Is it necessary that it's configurable?

File: README.md (hunk 1 above)
Comment (lebe1) [reply README.md:105]: Correct! This worked inside my container!

**Comment by lebe1** [comment]
> > I have one final question/thought:
> > ```
> > 1. Shouldn't we add the new keycloak environment variables to our `.env.examples` like 'KEYCLOAK_CUSTOMER_GROUP' ?
> >    Besides this, I could even see the two testing versions in our keycloak GUI being created, which is really nice! :)
> >    image
> > ```
> 
> maybe we should do this, but who will change it? Is it necessary that it's configurable?

I don't think it is necessary that it is configurable and it also might be that this variable won't ever be changed again, which probably goes along for a few other variables in the `.env.example`.
But since this variable is defined in the configs and every other variable is in the `.env.example` I think by being consistent and having each variable in the `.env.example` is better practice... ?

**Comment by nanu-c** [comment]
I have added an issue for this #156

**Pull request merged**